src/abstractdataframe/abstractdataframe.jl

```diff
@@ -410,6 +410,29 @@ function Base.isequal(df1::AbstractDataFrame, df2::AbstractDataFrame)
     return true
 end
 
+function Base.isapprox(df1::AbstractDataFrame, df2::AbstractDataFrame;
+                  atol::Real=0, rtol::Real=atol>0 ? 0 : √eps(),
+                  nans::Bool=false, norm::Function=abs)
+    size(df1, 2) == size(df2, 2) || return false
+    isequal(index(df1), index(df2)) || return false
+    for idx in 1:size(df1, 2)
+        col_1 = df1[!, idx]
+        col_2 = df2[!, idx]
+        ismissing(col_1 == col_2) && return missing
+        if !(col_1 isa AbstractVector{<:Number})  
+            isequal(col_1, col_2)
+        else
+            isapprox(col_1, col_2; atol, rtol, nans, norm) || return false
+            # if !(col_1 isa AbstractVector{<:Union{Missing, Number}}) # should we check for both || !df2[!, idx] isa AbstractVector{<:Union{Missing, Number}} 
+            #     ismissing(col_1 == col_2) && return missing
+            #     isequal(col_1, col_2) || return false
+            # else
+            #     ismissing(col_1 == col_2) && return missing
+            #     col_1 == col_2 || norm.(col_1-col_2) <= max.(atol, rtol*max.(norm.(col_1), norm.(col_2))) || (nans && isnan.(col_1) && isnan.(col_2)) || return false
+        end
+    end
+    return true
```

Comment:
```suggestion
    return all(isapprox.(eachcol(df1), eachcol(df2), atol=atol, rtol=rtol, nans=nans, norm=norm))
```

Reply:
Sorry for changing my mind. I was thinking what we should do and came to the conclusion we should be defensive. I.e. just fall back to `isapprox` from Base. If in the future Base starts to be more flexible then we will become more flexible automatically. Also now we error on "wrong" input, so we always can switch to not-error in a non-breaking way.

Note that I evaluate `isapprox` eagerly on purpose - to force an error if some columns are invalid.

In general - in order to use `isapprox` now the user should select numeric only columns in a data frame when doing a comparison which I think is not that bad.

Reply:
Thanks for the suggestion! :)
But one issue that still remains is the `abs` function that we use as default in the function definition

Reply:
Ah. Seems valid. I am kind of thinking towards possible improvements that can directly go into the Base method for arrays from the things we have observed here for DataFrames. Like it's abit incomplete to check the strings  only for equalty where a check for hamming distance can bring it in line with the check for Numbers.

Reply:
Ah - sorry for not commenting on this. I think we should keep `norm` as it is used in Julia Base.
This will require a bit more work as you need:
1. add LinearAlgebra to Project.toml
2. add import LinearAlgebra: norm in the DataFrames.jl file

Reply:
> that can directly go into the Base

I think you can just open an issue in Base and CC @nalimilan and me to discuss it there. As I have said - if it goes into Base then automatically we will also have it in DataFrames.jl for free. Thank you!

Reply:
Sorry for pestering you this way, but it seems like the Base methods will throw an error for `missing` values. Now it's alittle tricky to handle the `missing` case with the condition `return all(isapprox.(eachcol(df1), eachcol(df2), atol=atol, rtol=rtol, nans=nans, norm=norm))`.

Reply:
Yes - Base methods will error on `missing` and this is what I think is safe to do. As if we also throw an error in this case we can change this behavior in the future to whatever we like without being breaking. When you open an issue in Base then the case of `missing` should be raised I think.